src/Pages/EditProjectForm.jsx

```diff
@@ -1,3 +1,128 @@
-export default function EditProjectForm() {
-  return;
+import React, { useEffect } from "react";
+import { useState } from "react";
+import { useParams, useHistory } from "react-router";
+
+export default function EditProjectForm({ categories }) {
+  const history = useHistory();
+  const { userId, userName, projectId } = useParams();
+  const targetProjectId = parseInt(projectId);
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+  const [goal, setGoal] = useState(0);
+  const [projectCategoryIds, setProjectCategoryIds] = useState([]);
+
+  useEffect(() => {
+    fetch(`${process.env.REACT_APP_API_URL}/projects/${projectId}`)
+      .then((res) => res.json())
+      .then((currentProject) => {
+        setTitle(currentProject.title);
+        setDescription(currentProject.description);
+        setGoal(currentProject.goal);
+
+        const categories = currentProject.categories;
+
+        setProjectCategoryIds(
+          categories.map((category) => {
+            return category.category.id;
+          })
+        );
+      });
+  }, [targetProjectId]);
+
+  const handleCategoriesCheckbox = (e) => {
+    const isChecked = e.target.checked;
+    const selectedCategoryId = parseInt(e.target.value);
+
+    if (isChecked) {
+      setProjectCategoryIds([...projectCategoryIds, selectedCategoryId]);
+    } else {
+      const filteredCategoryIds = projectCategoryIds.filter(
+        (categoryId) => selectedCategoryId !== categoryId
+      );
+      setProjectCategoryIds(filteredCategoryIds);
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const projectDetails = {
+      title,
+      description,
+      goal: parseInt(goal),
+      categoryIds: projectCategoryIds,
+      userId: parseInt(userId),
+    };
+
+    const fetchOptions = {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(projectDetails),
+    };
+
+    fetch(
+      `${process.env.REACT_APP_API_URL}/projects/${projectId}`,
+      fetchOptions
+    )
+      .then((res) => res.json())
+      .then((editProject) => {
+        history.push(`/user/${userId}/${userName}/myProjects`);
```

Comment:
This is an interesting way to use routing but it seems redundant, is there a reason you're using parameters to pass around the userName *and* the user ID and not just passing down as props?

No need to change this now, just something to think about